src/addons/devkit/addon.cpp

```diff
@@ -82,6 +82,56 @@ std::atomic_bool shaders_pane_show_compute_shaders = true;
 uint32_t skip_draw_count = 0;
 std::atomic_uint32_t device_data_index = 0;
 
+// Per-shader snapshot data
+struct SnapshotData {
+  std::vector<int> occurrences;                      // position -> draw index
+  std::unordered_map<int, int> draw_index_positions; // draw index -> position
```

Comment:
Having a vector of occurrences in order, and an unordered map of <index, position> is a bit of an anti-pattern. Makes more sense to make a single sorted map. You can always iterate and get last of a map.